### DISCUSSION TITLE
Clarity of CLA

### DISCUSSION THREAD

**Discussion by ckipp01** [sbt/sbt#7048] [Q&A] [answered]
Hello! I recently noticed the following message about a pr of mine that was backported [here](https://github.com/sbt/sbt/pull/7039#issuecomment-1264587874).

> Hi @ckipp01,
>
> Thank you for your contribution! We really value the time you've taken to put this together.
>
> We see that you have signed the Lightbend Contributors License Agreement before, however, the CLA has changed since you last signed it.
> Please review the new CLA and sign it before we proceed with reviewing this pull request:
>
> https://www.lightbend.com/contribute/cla

With recent events at lightbend I'm a bit more cautions with CLA's and what I'm signing and it made me realize that I don't fully understand this CLA for sbt. From my understanding sbt is no longer being maintained by lightbend. I see that signing the lightbend CLA is still mentioned in the [CONTRIBUTING](https://github.com/sbt/sbt/blob/1.8.x/CONTRIBUTING.md#signing-the-cla) doc and also checked by CI. Why is signing the Lightbend CLA a requirement for the sbt project?

**Comment by eed3si9n** [comment]
@xuwei-k has also mentioned this recently - https://twitter.com/not_xuwei_k/status/1573869441187708928, specifically he has asked to remove the CLA (translation by me).

Although I'm the most active maintainer of sbt and its various submodules, technically speaking the copyright and legal ownership of them are under Lightbend, inc.

**Reply by ckipp01** [reply]
> Although I'm the most active maintainer of sbt and its various submodules, technically speaking the copyright and legal ownership of them are under Lightbend, inc.

Maybe a naive question, but in practice, what does this mean?

Could lightbend change the licensing of these modules?

> specifically he has asked to remove the CLA (translation by me).

This is sort of what I was indirectly asking as well. What would need to happen to remove this specific CLA, with the assumption that you (as the primary maintainer) would also be willing to do so?

**Reply by eed3si9n** [reply]
Yes, the benefit of waiving individual copyright to EPFL, ASF, or Lightbend is that they can unilaterally open or close the licensing etc as they see fit. I personally think it's highly unlikely that sbt, Zinc, etc licensing to change in the future since at that point I can fork. But in general, it's totally fair for me, you, or Yoshida-san as contributors to ask if they would drop CLA or transfer the ownership to some other body like EPFL or Scala Center.

**Reply by eed3si9n** [reply, marked as answer]
Looks like we did it cc @xuwei-k 
https://github.com/sbt/sbt/pull/7306

**Comment by nafg** [comment]
I believe Play is removing it.

And @octonato told me wrt. Slick,

> I think you should just remove the CLA check. For a few reasons...
> 1. Lightbend is not the steerer of the project anymore and it doesn't make sense for Slick to keep having it
> 2. The recent CLA changes related with the new Akka license. That's why users are requires to sign it again. Slick contributors should not be bothered by that.
>
> [...]
> your call if you add another one or leave without
> there many projects without CLA, but it's a nice to have. Just extra protection in case a company comes later saying that proprietary code was added without their authorization

Is Lightbend involved in SBT any more? I would think the same reasoning should apply.

**Reply by nafg** [reply]
P.S. If you have a suggestion for a replacement for Slick please share (but not here, that would be off topic)

**Comment by ckipp01** [comment]
Just wanted to through an update up on this. There was a recent discussion about this at the last Scala Center advisory board meeting and there has already been talks started between the Scala Center and Lightbend about this. No concrete results yet, but know that the correct people are having discussions about this.
